pkg/terra/relay.go

```diff
@@ -100,13 +102,15 @@ func (r *Relayer) Healthy() error {
 	return r.chainSet.Healthy()
 }
 
-func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2Provider, error) {
+// NewOCR2Provider creates a new OCR2ProviderCtx instance.
+func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2ProviderCtx, error) {
 	spec, ok := s.(OCR2Spec)
 	if !ok {
 		return nil, errors.New("unsuccessful cast to 'terra.OCR2Spec'")
 	}
 
-	chain, err := r.chainSet.Chain(spec.ChainID)
+	// At the time of creating services, we don't have a context, hence using Background context here.
+	chain, err := r.chainSet.Chain(context.Background(), spec.ChainID)
```

Comment:
Mmm interesting. I wonder if we really ought to pass a context in here - or perhaps have a lifecycle context that lives on the delegate?

Reply:
I had to change `Chain` method to receive a context, but for this constructor specifically, I would call @jmank88 or anybody else who understand terra package better than me. I am open to change this further.

Reply:
I think Sam's idea makes sense. An internal ctx could be passed that is cancelled when the the chainset is closed. On the other hand, we may eventually want a ctx added to `NewOCR2Provider` anyways - that could be messy or out of scope for now though.

Reply:
Implemented the proposed changes.